ext/proj4_c_impl/extconf.rb

```diff
@@ -12,9 +12,12 @@
 
   require "mkmf"
 
-  header_dirs_ =
-    [
-      ::RbConfig::CONFIG["includedir"],
+  with_default_header_paths = with_config("default-header-paths", default = true)
+  with_default_lib_paths = with_config("default-lib-paths", default = true)
```

Comment:
The trick is nice, but IMHO it getts confusing as this is not ruby syntax at all!
```suggestion
  with_default_header_paths = with_config("default-header-paths", true)
  with_default_lib_paths = with_config("default-lib-paths", true)
```